src/android/toga_android/app.py

```diff
@@ -62,6 +67,86 @@ def onActivityResult(self, requestCode, resultCode, resultData):
         except KeyError:
             print("No intent matching request code {requestCode}")
 
+    def onOptionsItemSelected(self, menuitem):
+        consumed = False
+        for cmd in self._impl.interface.commands._commands.union(self._impl.interface.main_window.toolbar._commands):
```

Comment:
AFAICT, this shouldn't be needed. Any command added to a toolbar should be automatically on the list of all commands; the toolbar is only there as a UI affordance to make the command easily accessible.

Also - is this the callback when a menu item is actually invoked? If it is... why are we iterating over all commands? Shouldn't we be setting up a mapping from `itemId` to `cmd` so we can do an O(1) lookup?

Reply:
> AFAICT, this shouldn't be needed.

Right! I'll remove the union

Reply:
> Shouldn't we be setting up a mapping from itemId to cmd ?

@freakboy3742 
Yes, that would be the other alternative. I chose to add the itemId as additional attribute of Command.
Would you prefer that I set up a lookup dictionary in `onPrepareOptionsMenu` which we could then use in `onOptionsItemSelected` ?

Reply:
I'll answer my question myself: It is better to store backend data (menuitem id) in a variable on the implementation layer rather than adding it to an object of the interface layer.
-> using lookup dictionary in the backend now